src/sentry/grouping/strategies/newstyle.py

```diff
@@ -364,7 +369,16 @@ def frame(frame, event, context, **meta):
         rv.update(contributes=False, hint="ignored due to recursion")
 
     if rv.tree_label:
-        rv.tree_label = {"datapath": frame.datapath, **rv.tree_label}
+        tree_label = {}
```

Comment:
Why is this change necessary?

Reply:
grouping components by default take the first tree label from their child component and propagate it. In the case of a frame this meant that only one tree-label from any of the frame's child components was propagated outside, so there can be only either `function` or `package` inside. (`_calculate_tree_label` in grouping/component.py)

This change here basically overrides the default tree label of `rv.tree_label = rv.values[0].tree_label` to instead merge all values's dictionaries into one.

This also causes the weirdness in the snapshots you pointed out because we used to use the function component's tree label even if the component was non-contributing (for some reason I haven't figured out, but that was a bug for sure)

(Note: Ideally at some point I'd like to refactor this entire mess such that tree labels are not propagated the way they are, I think I'll do that in a followup PR

Reply:
Makes sense.